### PR TITLE
refactor: Move off of tsdx and related tooling

### LINKED ISSUE #31: ci: move off of tsdx
Evict the project from txdx and switch the build process to use vite or rollup

### PR DESCRIPTION
**Pull request by cephalization**
Resolves #31 

- Remove direct tsdx, babel, jest, tslib deps
- Install vite, vitest, plugins
- Configure vite and vitest
- Update github workflows to use pnpm and node 20

TODO:

- [x] Upgrade storybook, install storybook vite plugin
- [x] Test new dist files in external project, ensure usage/imports remain the same

### CODE REVIEW

**Comment by cephalization** [comment]
It would be helpful to implement https://github.com/stackblitz-labs/pkg.pr.new for this PR so that we can test npm installs of point cloud from other apps:

- [ ] Are imports well-typed
- [ ] Are all desired files included in the pkg download
- [ ] Does esm vs cjs detection work correctly during import

**Comment by mikeldking** [comment]
very cool - we should do a test publish.

LGTM tho

**Comment by cephalization** [comment]
Published `@arizeai/point-cloud@4.0.0-0` / `@arizeai/point-cloud@4.0.0-0` and installed it in phoenix as well as another project with no issues. Imports are well typed and no runtime errors occur.

File: vite.config.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+});
+
+export default mergeConfig(
+    defineConfig({
+        plugins: [react()],
+        build: {
+            lib: {
+                // Could also be a dictionary or array of multiple entry points
+                entry: resolve(__dirname, "src/index.tsx"),
+                name: "PointCloud",
```
Comment (mikeldking) [inline vite.config.ts:20]: Does this do anything?

File: vite.config.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                // the proper extensions will be added
+                fileName: "index",
+                formats: ["es", "cjs"],
+            },
+            minify: false,
+            rollupOptions: {
+                // make sure to externalize deps that shouldn't be bundled
+                // into your library
+                external: [
+                    "react",
```
Comment (mikeldking) [inline vite.config.ts:30]: Weird that you need to do this explicitly...

**Review by mikeldking — APPROVED** [review]

File: vite.config.ts (hunk 2 above)
Comment (cephalization) [reply vite.config.ts:30]: The default vite behavior is to bundle all deps with the library. I don't really like it either

File: vite.config.ts (hunk 1 above)
Comment (cephalization) [reply vite.config.ts:20]: I learned today that this was optional, I will remove

File: vite.config.ts (hunk 2 above)
Comment (cephalization) [reply vite.config.ts:29]: Removed extraneous config for umd modules as well

**Pull request merged**